src/components/VideoThumbnail/VideoThumbnail.tsx

```diff
@@ -56,10 +72,37 @@ export function VideoThumbnail({
     );
   }
 
+  const progressValue = calculateProgress(videoLength, videoProgress);
+  const progressValuePercents = Math.round(progressValue * 100);
```

Comment:
Should these calculations only run when needed? (for example, by wrapping them in a condition that checks `showVideoProgress` is true)